### PR TITLE
TOML null handling [Adds #253]

### PR DESCRIPTION
**Pull request by yawkat**

### CODE REVIEW

**Comment by yawkat** [comment]
Oops, pushed the branch onto the main repo, I'll delete it later :)

File: toml/src/main/java/com/fasterxml/jackson/dataformat/toml/TomlGenerator.java [hunk 1]
```diff
@@ -570,7 +573,12 @@ public void writeNumber(String encodedValue) throws IOException {
 
     @Override
     public void writeNull() throws IOException {
-        throw new UnsupportedOperationException("Nulls are not supported by TOML");
+        if (TomlWriteFeature.FAIL_ON_NULL_WRITE.enabledIn(_tomlFeatures)) {
+            throw new TomlStreamWriteException("TOML null writing disabled (FAIL_ON_NULL_WRITE)", this);
```
Comment (cowtowncoder) [inline toml/src/main/java/com/fasterxml/jackson/dataformat/toml/TomlGenerator.java:577]: Maybe add `TomlWriteFeature.` prefix in error message to give context (bit easier for users to find)?

File: toml/src/main/java/com/fasterxml/jackson/dataformat/toml/TomlWriteFeature.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package com.fasterxml.jackson.dataformat.toml;
+
+import com.fasterxml.jackson.core.FormatFeature;
+
+/**
+ * Enumeration that defines all togglable features for TOML generators.
+ */
+public enum TomlWriteFeature implements FormatFeature {
+    FAIL_ON_NULL_WRITE(false);
```
Comment (cowtowncoder) [inline toml/src/main/java/com/fasterxml/jackson/dataformat/toml/TomlWriteFeature.java:9]: Javadoc would be nice (same for one feature in `TomlReadFeature`)

**Comment by cowtowncoder** [comment]
Cool. I am working on making `jackson-benchmarks` 2.13 have TOML read/write tests too and this will help resolve one existing issue!

**Comment by yawkat** [comment]
both comments implemented. Also added the doc for TomlReadFeature in a separate commit.

**Comment by cowtowncoder** [comment]
LGTM. I'll let you merge this, if and when it is ready & merge to `master`. I'll then play with the benchmarks bit more.
And may add some more tests; for some reason I think I am getting empty content with TOMLGenerator, but not yet sure why.

Also: I think there is something in generator that is not closing contexts appropriately, leading to concatenation of field names (as if close-object did not truncate name or something).

**Comment by yawkat** [comment]
okay, will do this merge and look at the generator tomorrow :)

**Comment by yawkat** [comment]
merged manually into 2.12 (776690f01d42ccda109b336875e686f063a6832b, 652769391279d86d28dc371741af94db202cee6b) merge into newer trees upcoming.
